### PR TITLE
Add comment to other observable instruments about repeated creation with callbacks

### PR DESCRIPTION
**Pull request by dashpole**
In https://github.com/open-telemetry/opentelemetry-go/pull/4820, I only added a comment describing the behavior to `Int64ObservableCounter`, but forgot other instruments.  This adds the comment to all observable instruments.

Fixes https://github.com/open-telemetry/opentelemetry-go/issues/5561

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
Is it not a problem that end users will not see this comment?

End users will just read this doc: https://pkg.go.dev/go.opentelemetry.io/otel/metric#Meter

Maybe we should add this kind of information to `sdk/metric` package documentation?

**Comment by dashpole** [comment]
I thought about adding it to the API, but this does seem like an (SDK) implementation-specific behavior.  I can add it to the package documentation.

**Review by dmitryax — APPROVED** [review]

**Comment by dashpole** [comment]
@pellared added to the SDK package documentation.  PTAL

**Review by MrAlias — APPROVED** [review]

File: sdk/metric/doc.go [hunk 1]
```diff
@@ -31,6 +31,14 @@
 // is being run on. That way when multiple instances of the code are collected
 // at a single endpoint their origin is decipherable.
 //
+// To avoid leaking memory, the SDK returns the same instrument for calls to
+// create new instruments with the same Unit, Name, and Description.
```
Comment (pellared) [inline sdk/metric/doc.go:35]: nit (use the same order as later)

```suggestion
// create new instruments with the same Name, Unit, and Description.
```

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**